lib/oidcstrategy.js

```diff
@@ -853,6 +859,40 @@ Strategy.prototype.setOptions = function setOptions(params, oauthConfig, options
  * @param {Function} next  -- when error occurs, call next(err)
  * @param {Function} callback
  */
+Strategy.prototype._decrytJWE = function decrytJWE(params, optionsToValidate, req, next, callback) {
+  const self = this;
+
+  var id_token = params.id_token;
+  var parts = id_token.split('.');
+
+  if (parts.length === 3)
+    return self._validateResponse(params, optionsToValidate, req, next, callback);
+  else if (parts.length !== 5)
```

Comment:
Logic would read easier with the 'else if' as 
else if (parts.length == 5) {
   do the decryption, then validate
}

return next (new Error('....');